internal/security/secretstoreclient/vault.go

```diff
@@ -171,21 +172,22 @@ func (vc *vaultClient) commonRequest(token *string, method string, path string,
 // delegate is a callback function called to handle the decoded response
 func (vc *vaultClient) processResponse(resp *http.Response, responseError error,
 	operationDescription string, expectedStatusCode int,
-	responseObject interface{}) (statusCode int, err error) {
+	responseObject interface{}) (int, error) {
 
 	if responseError != nil {
-		vc.logger.Error(fmt.Sprintf("unable to make request to %s failed: %s", operationDescription, err.Error()))
-		return 0, err
+		vc.logger.Error(fmt.Sprintf("unable to make request to %s failed: %s", operationDescription, responseError.Error()))
+		return 0, responseError
 	}
 	defer resp.Body.Close()
 
 	if resp.StatusCode != expectedStatusCode {
-		vc.logger.Error(fmt.Sprintf("request to %s failed with status: %s", operationDescription, resp.Status))
-		return resp.StatusCode, err
+		msg := fmt.Sprintf("request to %s failed with status: %s", operationDescription, resp.Status)
+		vc.logger.Error(msg)
+		return resp.StatusCode, errors.New(msg)
```

Comment:
just use `fmt.Errorf(...)` for returning error

Reply:
Change to fmt.Errorf() and logging err.Error() instead.